app/models/plugins.rb

```diff
@@ -10,13 +10,17 @@ def basic_create_for_page(plugin, page)
       plugin.save
     end
 
+    def class_from_name(plugin_name)
+      "Plugins::#{plugin_name.camelcase}".constantize
+    end
+
     def create_for_page(plugin_name, page, ref)
       return true if plugin_name.blank? || page.blank?
-      plugin_class = "Plugins::#{plugin_name.camelcase}".constantize
+      plugin_class = class_from_name(plugin_name)
       existing = plugin_class.where(ref: ref, page_id: page.id)
       return true unless existing.empty?
       plugin = plugin_class.new(plugin_class.const_get(:DEFAULTS))
-      plugin.page = page
+      plugin.page_id = page.id
```

Comment:
If I didn't, then it could get stuck in a callback loop. For some reason `plugin.page = page` was making the save hook on page get called.